graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+        return df
+
+    def to_dict(self) -> Dict[str, str]:
+        """Return a dictionary of sequences indexed by chains.
+
+        :return: Dictionary of chain-sequence mappings.
+        :rtype: Dict[str, str]
+        """
+        return (
+            self.df[["id", "sequence"]].set_index("id").to_dict()["sequence"]
+        )
+
+    def to_fasta(self, filename: str):
+        """Write the dataset to a FASTA file (indexed by chain id).
+
+        :param filename: Name of the output FASTA file.
+        :type filename: str
+        """
+        with open(filename, "w") as f:
+            for k, v in self.to_dict().items():
+                f.write(f">{k}\n")
+                f.write(f"{v}\n")
+
+    def standard_alphabet(self, update: bool = False):
+        """
+        Remove sequences with non-standard characters.
+
+        :param update: Whether to update the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame containing only sequences with standard characters.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[
+            self.df.sequence.map(
+                lambda x: set(x).issubset(set("ACDEFGHIKLMNPQRSTVWY"))
+            )
+        ]
+        if update:
+            self.df = df
+        return df
+
+    def download(
+        self,
+        out_dir=".",
+        overwrite: bool = False,
+        max_workers: int = 8,
+        chunksize: int = 32,
+    ):
+        """Download PDB files in the current selection.
+
+        :param out_dir: Output directory, defaults to ``"."``
+        :type out_dir: str, optional
+        :param overwrite: Overwrite existing files, defaults to ``False``.
+        :type overwrite: bool, optional
+        :param max_workers: Number of processes to use, defaults to ``8``.
+        :type max_workers: int, optional
+        :param chunksize: Chunk size for each worker, defaults to ``32``.
+        :type chunksize: int, optional
+        """
+        log.info(f"Downloading {len(self.unique_pdbs)} PDB files...")
+        download_pdb_multiprocessing(
+            self.unique_pdbs,
+            out_dir,
+            overwrite=overwrite,
+            max_workers=max_workers,
+            chunksize=chunksize,
+        )
+
+    def write_chains(self) -> List[Path]:
+        """Write chains in current selection to disk. e.g., we create a file
+        of the form ``4hbb_A.pdb`` for chain ``A`` of PDB file ``4hhb.pdb``.
+
+        If the PDB files are not contained in ``self.pdb_dir``, they are
+        downloaded.
+
+        :return: List of paths to written files.
+        :rtype: List[Path]
+        """
+        # Get dictionary of PDB code : List[Chains]
+        df = self.df.groupby("pdb")["chain"].agg(list).to_dict()
+
+        # Check we have all source PDB files
+        downloaded = os.listdir(self.pdb_dir)
+        downloaded = [f for f in downloaded if f.endswith(".pdb")]
+
+        if to_download := [
```

Comment:
I think this walrus will break 3.7 compatibility

Reply:
Addressed in [this commit](https://github.com/a-r-j/graphein/pull/272/commits/3b42dab1be3429af2e9b2c5da59166bc722c8046).